src/jsx/Configuration.js

```diff
@@ -25,103 +31,280 @@ const Configuration = (props) => {
   const socketContext = useContext(SocketContext);
 
   // React State
-  const [edfFile, setEdfFile] = useState({});
-  const [edfType, setEdfType] = useState('iEEG');
-  const [eventsTSV, setEventsTSV] = useState({});
-  const [bidsDirectory, setBidsDirectory] = useState(null);
-  const [lineFreq, setLineFreq] = useState('');
-  const [siteID, setSiteID] = useState('');
-  const [projectID, setProjectID] = useState('');
-  const [subProjectID, setSubProjectID] = useState('');
-  const [visitLabel, setVisitLabel] = useState('');
-  const [headerFields, setHeaderFields] = useState(null);
-  const [edfHeader, setHeader] = useState({
+  const state = {};
+  state.edfFiles = {};
+  [state.edfFiles.get, state.edfFiles.set] = useState([]);
```

Comment:
Hi @laemtl, I'm unsure if you need to be initiating `state.edfFiles = {};` when the `useState({});` should be setting the initial type to an object. Similar for all the other lines below. Is that necessary if you combine everything into one giant `state` object?

Reply:
I need to initialize state.edfFiles to {} in order to add the get and set keys without error/warning. useState set the initial value of the set key, not the parent dictionary.

Reply:
I combine everything into a common state object in order to be able to loop on the keys to call setTask on all entries (previously the task keys were initially null).